metrics-server/src/lib.rs

```diff
@@ -31,10 +32,19 @@ mod server;
 mod tokio_runtime;
 mod tokio_task;
 
+/// Monitor (metrics) for a Nimiq task
 #[derive(Clone)]
 pub struct NimiqTaskMonitor {
-    pub name: String,
-    pub monitor: TaskMonitor,
+    name: String,
+    monitor: TaskMonitor,
+}
+
+impl NimiqTaskMonitor {
+    pub fn new(name: String, monitor: TaskMonitor) -> Self {
+        // Metrics name **must** be snake case according to OpenMetrics spec.
+        let name = name.to_case(Case::Snake);
```

Comment:
Instead of fixing up the input, I'd suggest to panic if the input contains a non-alphanumeric non-underscore character.

This would not require the new dependency.